polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisTreeMapMetaStoreSessionImpl.java

```diff
@@ -448,13 +448,16 @@ public int lookupEntityGrantRecordsVersion(
   /** {@inheritDoc} */
   @Override
   public @NotNull PolarisPrincipalSecrets generateNewPrincipalSecrets(
-      @NotNull PolarisCallContext callCtx, @NotNull String principalName, long principalId) {
+      @NotNull PolarisCallContext callCtx,
+      @NotNull String principalName,
+      long principalId,
+      PrincipalSecretsGenerator generator) {
```

Comment:
Same here - rather than passing in the generator as an argument, I think it should be a constructor argument

Reply:
reworked to avoid modifying public interfaces.